packages/trader/src/Modules/Trading/Components/Form/TradeParams/__tests__/strike.spec.tsx

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { mockStore } from '@deriv/stores';
+import { isMobile, isDesktop, VANILLALONG } from '@deriv/shared';
+import TraderProviders from '../../../../../../trader-providers';
+import Strike from '../strike';
+
+const mocked_store = {
+    modules: {
+        trade: {
+            barrier_1: '1',
+            barrier_choices: ['16', '33', '40'],
+            duration_unit: 'm',
+            onChange: jest.fn(),
+            validation_errors: {},
+            expiry_type: 'endtime',
+            expiry_date: null,
+            vanilla_trade_type: VANILLALONG.CALL,
+        },
+    },
+};
+const mocked_strike_param_modal = 'Mocked Strike Param Modal Component';
+const strike_price = 'Strike price';
+const strike_prices = 'Strike Prices';
+const spot = 'Spot';
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isMobile: jest.fn().mockReturnValue(false),
+    isDesktop: jest.fn().mockReturnValue(true),
+}));
+jest.mock('Modules/Trading/Containers/strike-param-modal', () => jest.fn(() => <div>{mocked_strike_param_modal}</div>));
+
+describe('<Strike />', () => {
+    const mockStrike = (store: typeof mocked_store) => {
+        return (
+            <TraderProviders store={mockStore(store)}>
+                <Strike />
+            </TraderProviders>
+        );
+    };
+
+    it('should render Strike component with default props', () => {
+        render(mockStrike(mocked_store));
+
+        expect(screen.getByText(strike_price)).toBeInTheDocument();
+    });
+
+    it('should not allow to change strike manually', () => {
+        render(mockStrike(mocked_store));
+
+        const inputField: HTMLInputElement = screen.getByTestId('dt_strike_input');
+        expect(inputField.value).toBe('1');
+
+        fireEvent.change(inputField, { target: { value: '2400.00' } });
```

Comment:
Sorry, will it be possible to use `userEvent`? More optional tip, but during last 'testing' sessions we have an agreement to use it instead of `fireEvent`.

Reply:
Let me try but I'm actually trying to make a test case which would check if it's not possible to change this field through different ways, including dev tools. Not sure if it's replicable with userEvent.

Reply:
Sure, if fireEvent is better for such purposes, then ignore my comment)

Reply:
I changed other occurrences  to user event, this one can't be fully replicated without fireEvent usage 👍🏻